src/pages/shared/ModlogPage.tsx

```diff
@@ -22,20 +22,34 @@ import { useParams } from "react-router";
 import { getHandle } from "../../helpers/lemmy";
 import { useBuildGeneralBrowseLink } from "../../helpers/routes";
 import { buildCommunityLink } from "../../helpers/appLinkBuilder";
+import { RootState, useAppSelector } from "../../store";
+import useCanModerate from "../../features/moderation/useCanModerate";
+import { useUserDetails } from "../../features/user/useUserDetails";
 
 export type ModlogItemType =
   GetModlogResponse[keyof GetModlogResponse] extends (infer T)[] ? T : never;
 
 export default function ModlogPage() {
-  const { community } = useParams<{ community?: string }>();
+  const { community, handle } = useParams<{
+    community?: string;
+    handle?: string;
+  }>();
 
-  if (!community) return <GlobalModlog />;
+  if (handle) return <ModlogByUserHandle handle={handle} />;
+  if (community) return <ModlogByCommunityName communityName={community} />;
 
-  return <ModlogByCommunityName communityName={community} />;
+  return <GlobalModlog />;
 }
 
+const userSelector = (state: RootState) =>
+  state.auth.site?.my_user?.local_user_view?.person;
+
 const GlobalModlog = memo(function GlobalModlog() {
-  return <ModlogByCommunity />;
+  const canModerate = useCanModerate(true);
+  const user = useAppSelector(userSelector);
```

Comment:
inline the selector function here

Reply:
The same selector function is used again below, on line 62

Reply:
Sure can!